src/org/usfirst/frc2130/visionTargetting/Robot.java

```diff
@@ -0,0 +1,263 @@
+	
+	public static double run;
+	
+
+	private static final int IMG_WIDTH = 320;
+	private static final int IMG_HEIGHT = 240;
+
+	private boolean run1;
+    
+    public static OI oi;
+    // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DECLARATIONS
+    public static robotDrive robotDrive;
+
+    // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DECLARATIONS
+
+    /**
+     * This function is run when the robot is first started up and should be
+     * used for any initialization code.
+     */
+    @Override
+    public void robotInit() {
+    RobotMap.init();  
+    	
+    // Creates UsbCamera and MjpegServer [1] and connects them
+
+    CameraServer.getInstance().startAutomaticCapture();
+
+    // Creates the CvSink and connects it to the UsbCamera
+    CvSink cvSink = CameraServer.getInstance().getVideo();
+
+    // Creates the CvSource and MjpegServer [2] and connects them
+    CvSource outputStream = CameraServer.getInstance().putVideo("Blur", 640, 480); 
+       
+         
+        // BEGIN AUTOGENERATED CODE, SOURCE = ROBOTBUILDER ID = CONSTRUCTORS
+        robotDrive = new robotDrive();
+
+        // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID = CONSTRUCTORS
+        // OI must be constructed after subsystems. If the OI creates Commands
+        //(which it very likely will), subsystems are not guaranteed to be
+        // constructed yet. Thus, their requires() statements may grab null
+        // pointers. Bad news. Don't move it.
+        oi = new OI();
+
+        // instantiate the command used for the autonomous period
+        // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=AUTONOMOUS
+
+        autonomousCommand = new AutonomousCommand();
+
+        // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=AUTONOMOUS
+        
+        CameraServer.getInstance().startAutomaticCapture();
+        UsbCamera camera = new UsbCamera("USB Camera", 0);
+        camera.setResolution(IMG_WIDTH, IMG_HEIGHT);
+        
+        CameraServer.getInstance().getVideo();
+       
+        new Thread(() -> {        
+            CvSink cvSink = CameraServer.getInstance().getVideo();
+            CvSource outputStream = CameraServer.getInstance().putVideo("Blur", 640, 480);
+            
+            Mat source = new Mat();
+            Mat output = new Mat();
+            
+            while(!Thread.interrupted()) {
+                cvSink.grabFrame(source);
+                Imgproc.cvtColor(source, output, Imgproc.COLOR_BGR2GRAY);
+                outputStream.putFrame(output);
+            }
+            
+            visionThread = new VisionThread(camera, new GripPipeline(), pipeline -> {
+                
+            	synchronized (imgLock) {
+        			if(!pipeline.filterContoursOutput().isEmpty()) {
+        				run1 = true;
+        			}else {
+        				run1 = false;
+        			}
+        			
+        			
+        		}
+            	
+                
+            });
+            visionThread.start();
```

Comment:
Lines 174-188 are the key lines that we want to preserve.  This section is actually creating yet another thread that is kicked off from the outer thread from before.  It only starts when the other thread is interrupted due to the while() loop above it.  Instead, we want to move it outside the other thread and remove the other code.

This thread behaves very much like the other thread except that it takes a camera instance, a pipeline instance, and a callback function to call when the pipeline is done processing.  Some of the code from above needs to be moved here if we want to see the outputs of the processing on the dashboard (not sure if that's desired or not).